src/components/EnvironmentOverride/SecretOverrides.tsx

```diff
@@ -82,14 +82,14 @@ export default function SecretOverrides({ parentState, setParentState, ...props
                 {secrets &&
                     Array.from(secrets)
                         .sort((a, b) => a[0].localeCompare(b[0]))
-                        .map(([name, { data, defaultData, global, ...rest }]) => {
+                        .map(([name, { data, defaultData, global, esoSecretData, secretData, ...rest }]) => {
                             return (
                                 <ListComponent
                                     key={name || Math.random().toString(36).substr(2, 5)}
                                     appChartRef={appChartRef}
                                     name={name}
                                     type="secret"
-                                    label={global ? (data ? 'modified' : '') : 'env'}
+                                    label={global ? (data || esoSecretData.secretStore || secretData ? 'modified' : '') : 'env'}
```

Comment:
Can we avoid nesting in the ternary operator and make a separate function?

Reply:
As per discussion we are not making changes